_zconf.py

```diff
@@ -235,7 +235,7 @@ def Wait_for_service_remove(self, t_seconds, target_service):
     while time.time() < t_end:
       services = self.l.removed_services()
       for service in services:
-        if target_service in service:
+        if target_service.name == service:
```

Comment:
- 'target_service' is a string while 'service' is a dictionary
- testing for equality might be too strict, checking for substring is recommended 

The following line of code should work
> if target_service in service.name: